lib/rubocop/cop/style/documentation.rb

```diff
@@ -47,18 +47,25 @@ def check(ast, ast_with_comments)
           end
         end
 
+        def module_definition?(node)
+          # 'Const = Class.new' or 'Const = Module.new' are module definitions
+          node.class_type? ||
+            node.module_type? ||
+            (node.casgn_type? &&
+              (rhs = node.children[2]).send_type? &&
```

Comment:
I don't like the usage of array references and `children` in general, as it really obscures what you're trying to do. Even I'm not sure about it, so people not familiar with `parser` will be extra confused. Individual node destructuring is more verbose, but also way easier to comprehend as can give sensible names to everything you care about. I'd also say that the entire boolean expression is pretty hard to comprehend/debug, so I'd split it into smaller chunks. 

It's also a bit debatable whether a class check is a good idea in something named `module_definition?`.